core/spam_api.php

```diff
@@ -0,0 +1,44 @@
+<?php
+# MantisBT - A PHP based bugtracking system
+
+# MantisBT is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# MantisBT is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with MantisBT.  If not, see <http://www.gnu.org/licenses/>.
+
+require_api( 'access_api.php' );
+require_api( 'history_api.php' );
+
+/**
+ * Checks if the current user is suspected to be a spammer.  This is useful to run before
+ * actions like adding issues or issue notes.  If the user is determined to demonstrated
+ * spammy behavior, then this method with trigger an error and exit the script.
+ */
+function spam_check() {
+	if( OFF == config_get_global( 'allow_signup' ) ) {
+		return;
+	}
+
+	if( access_get_global_level() > config_get( 'default_new_account_access_level' ) ) {
+		return;
+	}
+
+	$t_spam_max_count = config_get( 'spam_max_event_count' );
+	if( $t_spam_max_count == 0 ) {
+		return;
+	}
+
+	if( history_count_recent_events( config_get( 'spam_time_windows_in_seconds' ) ) <= $t_spam_max_count ) {
+		return;
+	}
+
```

Comment:
Revised to pass parameters to include the parameters, but kept it in seconds.  If we allow seconds resolution, then we want to display it correctly in error messages.